src/platform_impl/ios/monitor.rs

```diff
@@ -175,8 +167,8 @@ impl fmt::Debug for MonitorHandle {
 
 impl MonitorHandle {
     pub fn retained_new(uiscreen: id) -> MonitorHandle {
+        assert_main_thread!("`MonitorHandle` can only be cloned on the main thread on iOS");
```

Comment:
This has nothing to do with that PR, but could `libdispatch` used to `drop` from the main thread? So we won't have runtime panics like that? Though I think the issue that you can't use that from the other thread as well?

It just all looks wrong to me. We say that winit window is `Send` and you can get `MonitorHandle` from the `Window`(owning one iirc), but when you drop it you'll panic?

Reply:
There is _so_ much thread-unsafety in our macOS (and probably also iOS, but less well-versed in that) implementation, really, you have no _idea_!

Anyway, to answer your question: Yes, we could and should use `libdispatch` to do things on the main thread, both here and in a lot of other places. I'm in the process of exploring solutions, have focused on `objc2` stuff first though.